cfme/v2v/infrastructure_mapping.py

```diff
@@ -0,0 +1,305 @@
+        return was_change
+
+    def after_fill(self, was_change):
+        self.next_btn.click()
+
+
+class InfrastructureMappingForm(InfrastructureMappingView):
+    """All infrastructure mapping View"""
+
+    title = Text(locator='.//h4[contains(@class,"modal-title")]')
+
+    @View.nested
+    class general(InfraMappingCommonButtons):  # noqa
+        name = TextInput(name="name")
+        description = TextInput(name="description")
+        name_help_text = Text(locator='.//div[contains(@id,"name")]/span')
+        description_help_text = Text(locator='.//div[contains(@id,"description")]/span')
+        fill_strategy = WaitFillViewStrategy()
+
+        def after_fill(self, was_change):
+            if was_change:
+                self.next_btn.click()
+
+    @View.nested
+    class cluster(ComponentView): # noqa
+        comp_name = "cluster"
+
+    @View.nested
+    class datastore(ComponentView): # noqa
+        comp_name = "datastore"
+
+    @View.nested
+    class network(ComponentView): # noqa
+        comp_name = "network"
+
+        def after_fill(self, was_change):
+            if self.create.is_displayed:
+                self.create.click()
+            elif self.save.is_displayed:
+                self.save.click()
+
+    @View.nested
+    class result(View):  # noqa
+        close = Button("Close")
+        continue_to_plan_wizard = Button("Continue to the plan wizard")
+        success_icon = Text('.//div[contains(@class,"wizard-pf-success-icon")]')
+
+    def after_fill(self, was_change):
+        if was_change:
+            self.result.close.click()
+
+
+class AllInfraMappingView(InfrastructureMappingView):
+    """All infrastructure mapping View"""
+
+    infra_mapping_list = InfraMappingList("infra-mappings-list-view")
+    create_infra_mapping = Text(locator='(//a|//button)[text()="Create Infrastructure Mapping"]')
+    configure_providers = Text(locator='//a[text()="Configure Providers"]')
+    paginator_view = View.include(V2VPaginatorPane, use_parent=True)
+    search_box = TextInput(locator=".//div[contains(@class,'input-group')]/input")
+    clear_filters = Text(".//a[text()='Clear All Filters']")
+    sort_order = Text(".//button[./span[contains(@class,'sort-direction')]]")
+    filter_by_dropdown = SelectorDropdown("id", "filterFieldTypeMenu")
+    sort_by_dropdown = SelectorDropdown("id", "sortTypeMenu")
+
+    @property
+    def is_displayed(self):
+        return len(self.browser.elements(".//div[contains(@class,'spinner')]")) == 0 and (
+            self.create_infra_mapping.is_displayed
+            or self.infra_mapping_list.is_displayed
+            or self.configure_providers.is_displayed
+        )
+
+
+class AddInfrastructureMappingView(InfrastructureMappingForm):
+    form_title_text = VersionPicker(
+        {
+            Version.lowest(): "Infrastructure Mapping Wizard",
+            "5.10": "Create Infrastructure Mapping",
+        }
+    )
+
+    @property
+    def is_displayed(self):
```

Comment:
Looks like this could be moved to `InfrastructureMappingForm.is_displayed`, since the child classes are both setting `form_title_text`